src/disk_io_thread.cpp

```diff
@@ -3100,7 +3101,6 @@ constexpr disk_job_flags_t disk_interface::cache_hit;
 		DLOG("started disk thread %s\n", thread_id_str.str().c_str());
 
 		std::unique_lock<std::mutex> l(m_job_mutex);
-		if (m_abort) return;
```

Comment:
even if we have aborted at this point, we still need to enter the main loop, process jobs and exit gracefully